### ISSUE TITLE
Can't install in Ubuntu 16.04

### ISSUE THREAD

**Issue by THPubs** [paradoxxxzero/gnome-shell-system-monitor-applet#309]
Installation fails in Ubuntu 16.04 with Ubuntu 3.18 with no error.

**Comment by Enigma0** [comment]
Can't install on 16.04 either. What is needed to get this working? It's the most useful Gnome extension around. Always assumed this had a huge following.

**Comment by THPubs** [comment]
Just tried to install it on Fedora 24 and it also didn't work :-(

**Comment by p91paul** [comment]
To help the maintainer understand the issue, you could do the following: 

* launch journalctl -f in a terminal
* install the extension and enable it.
* when it fails, get back to your opened terminal, stop journalctl with Ctrl-C, and post the produced output here.

**Comment by THPubs** [comment]
I just ran `journalctl -f` and it gave me this :

    -- Logs begin at Sat 2016-07-02 06:29:24 IST. --
    Jul 02 06:48:05 pc1 /usr/lib/snapd/snapd[895]: daemon.go:181: DEBUG: uid=1000;@ GET /v2/snaps?q=fir 1.012622512s 200
    Jul 02 06:48:05 pc1 /usr/lib/snapd/snapd[895]: daemon.go:181: DEBUG: uid=1000;@ GET /v2/snaps?q=fire 1.034650647s 200
    Jul 02 06:48:05 pc1 /usr/lib/snapd/snapd[895]: daemon.go:181: DEBUG: uid=1000;@ GET /v2/snaps/shadowsocks 50.279µs 404
    Jul 02 06:48:05 pc1 gnome-session[1664]: (gnome-software:1829): Gs-WARNING **: failed to call gs_plugin_refine on snappy: snapd returned status code 404: Not Found
    Jul 02 06:48:05 pc1 gnome-session[1664]: 2 Jul 06:48:05 - Serve proxy.pac
    Jul 02 06:48:13 pc1 gnome-session[1664]: [1:1:0702/064813:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 06:48:13 pc1 gnome-session[1664]: [1:1:0702/064813:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 06:48:14 pc1 gnome-session[1664]: [1:1:0702/064814:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 06:48:15 pc1 dleyna-server-service[5557]: dLeyna: Exit
    Jul 02 06:48:17 pc1 gnome-session[1664]: (gnome-shell:1728): GLib-GObject-WARNING **: /build/glib2.0-7IO_Yw/glib2.0-2.48.1/./gobject/gsignal.c:2635: instance '0x9d1c860' has no handler with id '1'


Then while it's running, I tried to install the plugin. The plugin did not install, just asked whether to install or not. First time I ran I didn't got any changes in the terminal. But the second time, I got this :

    -- Logs begin at Sat 2016-07-02 06:29:24 IST. --
    Jul 02 07:13:54 pc1 /usr/lib/snapd/snapd[895]: daemon.go:181: DEBUG: uid=1000;@ GET /v2/snaps/atom-cwayne 33.007µs 404
    Jul 02 07:13:54 pc1 gnome-session[1664]: (gnome-software:1829): Gs-WARNING **: failed to call gs_plugin_refine on snappy: snapd returned status code 404: Not Found
    Jul 02 07:13:55 pc1 gnome-session[1664]: [119B blob data]
    Jul 02 07:13:55 pc1 gnome-session[1664]: [119B blob data]
    Jul 02 07:13:57 pc1 gnome-session[1664]: 2 Jul 07:13:57 - Serve proxy.pac
    Jul 02 07:14:04 pc1 dleyna-server-service[6064]: dLeyna: Exit
    Jul 02 07:14:07 pc1 gnome-session[1664]: (gnome-shell:1728): GLib-GObject-WARNING **: /build/glib2.0-7IO_Yw/glib2.0-2.48.1/./gobject/gsignal.c:2635: instance '0x1b06c50' has no handler with id '1'
    Jul 02 07:14:11 pc1 gnome-session[1664]: [1:1:0702/071411:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:14:12 pc1 gnome-session[1664]: [1:1:0702/071412:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:14:42 pc1 gnome-session[1664]: [1:1:0702/071442:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:15:41 pc1 gnome-session[1664]: [1:1:0702/071541:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:15:42 pc1 gnome-session[1664]: [1:1:0702/071542:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:15:55 pc1 gnome-session[1664]: [1:1:0702/071555:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:15:56 pc1 gnome-session[1664]: [1:1:0702/071556:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:16:14 pc1 gnome-session[1664]: (gnome-shell:1728): GLib-GObject-WARNING **: /build/glib2.0-7IO_Yw/glib2.0-2.48.1/./gobject/gsignal.c:2635: instance '0xbcedb30' has no handler with id '1'
    Jul 02 07:16:14 pc1 gnome-session[1664]: [1:1:0702/071614:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:16:14 pc1 gnome-session[1664]: [1:1:0702/071614:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:16:52 pc1 gnome-session[1664]: [1:1:0702/071652:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:16:52 pc1 gnome-session[1664]: [1:1:0702/071652:ERROR:PlatformKeyboardEvent.cpp(117)] Not implemented reached in static PlatformEvent::Modifiers blink::PlatformKeyboardEvent::getCurrentModifierState()
    Jul 02 07:17:01 pc1 CRON[6444]: pam_unix(cron:session): session opened for user root by (uid=0)
    Jul 02 07:17:01 pc1 CRON[6445]: (root) CMD (   cd / && run-parts --report /etc/cron.hourly)
    Jul 02 07:17:01 pc1 CRON[6444]: pam_unix(cron:session): session closed for user root
    Jul 02 07:17:06 pc1 gnome-session[1664]: (gnome-shell:1728): GLib-GObject-WARNING **: /build/glib2.0-7IO_Yw/glib2.0-2.48.1/./gobject/gsignal.c:2635: instance '0xc0cd760' has no handler with id '1'

**Comment by p91paul** [comment]
Unfortunately, doesn't look like anything relevant happened. Please repeat the test, but after the install, restart gnome-shell (alt-f2, r, enter). Then post the output produced by journalctl.

**Comment by kensmith77** [comment]
This is all I could find relevant from journalctl:
```
Jul 14 12:19:59 nexus org.gnome.Shell.desktop[1241]: Gjs-Message: JS LOG: System monitor applet init from /usr/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com
Jul 14 12:19:59 nexus org.gnome.Shell.desktop[1241]: Gjs-Message: JS LOG: Extension "system-monitor@paradoxxx.zero.gmail.com" had error: TypeError: Background.from_string is not a function
```

**Comment by p91paul** [comment]
@kensmith77 This is it! At least there is now something to look for :)

**Comment by joadr** [comment]
worked for me after manually installing it, but, got this message on journalctl but shouln't be relevant, isn't it?

oct 19 15:09:59 ubbox gnome-session[2433]: Gjs-Message: JS WARNING: [/home/joaco/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js 872]: anonymous function does not always return a value
oct 19 15:09:59 ubbox gnome-session[2433]: Gjs-Message: JS LOG: System monitor applet init from /home/joaco/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com
oct 19 15:09:59 ubbox gnome-session[2433]: Gjs-Message: JS LOG: System monitor applet enabling
oct 19 15:09:59 ubbox gnome-session[2433]: Gjs-Message: JS LOG: System monitor applet enabling done

**Comment by franglais125** [comment]
@joadr That's just an innocuous error. You can modify line 866 in extension.js to avoid it:
Here: /home/joaco/.local/share/gnome-shell/extensions/system-monitor@paradoxxx.zero.gmail.com/extension.js

Line 866: `return;` modify to `return false;`

**Comment by chrisspen** [comment]
It looks like @paradoxxxzero is no longer using or maintaining this project. I've forked this and added @franglais125 patch, and now this works for me perfectly on Ubuntu 16.04/Gnome 3.18.2.

**Comment by p91paul** [comment]
If you are really willing to maintain this extension, please contact @paradoxxxzero or @darkxst. I'm sure they would be glad to hand this over to someone with enough free time to keep it running; plus, on extensions.gnome.org extension ownership can be transferred by admins. The author's permission is appreciated by reviewers considering a transfer.

**Comment by Aaronik** [comment]
@chrisspen, Installing your fork got the applet working for me for the first time since I installed Ubuntu 16.04. Thank you so much! I won't presume to do it over your head, but please do consider linking your fork in this issue, and possibly consider making a PR either with your changes, or with a link to your fork in the README of this repo. Thanks again!

**Comment by chrisspen** [comment]
@Aaronik I've already submitted a pull request. Unfortunately, it looks like the developer has vanished, so it's unlikely it or the other 19 requests will ever be addressed.

**Comment by paradoxxxzero** [comment]
I'm very sorry for my lack of presence here, I transmitted managing rights
to other devs a long time ago but it seems like they are gone. Would you
like to take over the maintenance of it?

Le ven. 9 déc. 2016 à 21:20, Chris Spencer <notifications@github.com> a
écrit :

**Comment by chrisspen** [comment]
Sure, it'd be my pleasure.

**Comment by paradoxxxzero** [comment]
Here you go, thanks!

**Comment by franglais125** [comment]
@chrisspen Well, get ready for an avalanche of pull-requests :D

And thanks for taking on this...

**Comment by darkxst** [comment]
On 10/12/16 10:42, Mounier Florian wrote:
> I'm very sorry for my lack of presence here, I transmitted managing rights
> to other devs a long time ago but it seems like they are gone.
I'm still around, just never seem to have time to maintain this anymore! 
Thanks for taking over Chrispen!

**Comment by chrisspen** [comment]
This should now be resolved by #335.
